tests/components/AugmentedResultList/AugmentedResultList.spec.ts

```diff
@@ -0,0 +1,219 @@
+            matchingIdField,
+            fetchAugmentData,
+            matchingFunction,
+        });
+
+        componentSetup = Mock.advancedComponentSetup(AugmentedResultList, testOptions);
+        return componentSetup;
+    };
+
+    const verifyAugmentedResultData = (resultData: any, results: IQueryResult[]) => {
+        resultData.forEach((data: any) => {
+            const matchingResult = results.find((result) => result.raw.id === data.id);
+            if (matchingResult) {
+                for (const key in data) {
+                    expect(matchingResult.raw[key]).toEqual(data[key]);
+                }
+            }
+        });
+    };
+
+    const verifyUntouchedResults = (resultData: {}[], results: IQueryResult[]) => {
+        const idString = matchingIdField.replace('@', '');
+
+        resultData.forEach((data: any) => {
+            const ids = resultData.map((data: any) => (data as any)[idString]);
+            const otherResults = results.filter((result) => !ids.find((id) => (result as any)[idString] === id));
+
+            otherResults.forEach((result: IQueryResult) => {
+                for (const key in data) {
+                    if (key !== idString) {
+                        expect(result.raw[key]).toBeUndefined;
+                    }
+                }
+            });
+        });
+    };
+
+    const verifyAugmentedCommonData = (commonData: any, results: IQueryResult[]) => {
+        results.forEach((result) => {
+            for (const key in commonData) {
+                expect(result.raw[key]).toEqual(commonData[key]);
+            }
+        });
+    };
+
+    const verifyAugmentedResults = (returnData: IPromiseReturnArgs<IAugmentData>, results: IQueryResult[]) => {
+        verifyAugmentedResultData(returnData.data.resultData, results);
+        verifyAugmentedCommonData(returnData.data.commonData, results);
+        verifyUntouchedResults(returnData.data.resultData, results);
+    };
+
+    beforeAll(() => {
+        Logger.disable();
+    });
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+        const endpoint = sandbox.createStubInstance(SearchEndpoint);
+        endpoint.search.callsFake(() => Promise.resolve(createFakeResultsThatMatch(10)));
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    afterAll(() => {
+        Logger.enable();
+    });
+
+    it('should augment results with object data', (done) => {
+        const numResults = 10;
+        const data = createFakeResultsThatMatch(numResults);
+        const test = createComponent(stubFetchAugmentData);
+
+        test.cmp.buildResults(data).then(() => {
+            expect(test.cmp.getDisplayedResults().length).toEqual(numResults);
+            verifyAugmentedResults(returnData, test.cmp.getDisplayedResults());
+            done();
+        });
+    });
+
+    it('should augment results with object data using provided matching function', (done) => {
```

Comment:
It would be nice to have a test to validate that precedence between common and specific data is applied properly.

Reply:
done 👍